### DISCUSSION TITLE
Does bleak support bluetooth pairing on the latest macOS (Ventura, 13.3.1)?

### DISCUSSION THREAD

**Discussion by cmjonze** [hbldh/bleak#1292] [Q&A]
From searching the discussion forms it looks like the answer is no, but I wanted to ask to be sure.  Thank you.

**Comment by dlech** [comment]
On macOS, if a characteristic requires authentication, the OS will prompt the user the first time the characteristic is used. There is no way to programatically initiate pairing/bonding or programatically provide the passphrase on macOS.

**Comment by cmjonze** [comment]
Not my question -- If you do try to use the characteristic and you're promted to pair, does it complete successfully using bleak and any of the bluetooth pairing methods?

**Reply by cmjonze** [reply]
Also, sorry if I'm not being clear:  I'm wanting the mac to pair and bond with a bluetooth peripheral.

**Reply by dlech** [reply]
As long as the user enters the correct passphrase and does not cancel, Bleak has no idea that anything unusual happened and it looks like a normal read or write. The pairng/bonding is handled completely by the OS and so it is completely transparent to Bleak.

**Reply by dlech** [reply]
As long as the connected device is compatible with macOS, it should work.
